### PR TITLE
Fix CRLF line-ending parsing for comments.

### LINKED ISSUE #25182: Rustdoc source code output dislikes windows line endings
Image says it all really, running rustdoc on files with windows style line-endings inserts (or doesn't strip?) additional newline characters, screwing up alignment with the line numbers:
image

This only seems to affect regular comments, doc comments are fine.

Edit: since it's not really obvious from the image due to the nature of the bug, it should look like this:
```rust
// Test comment
pub fn test() {}

// Test comment
pub fn test2() {}

// Test comment
pub fn test3() {}
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes #25182, parser didn't account for \r\n in regular comments, only doc-comments.

### CODE REVIEW

File: src/libsyntax/parse/lexer/mod.rs [hunk 1]
```diff
@@ -403,45 +403,47 @@ impl<'a> StringReader<'a> {
...
-                        });
+                            })
+                        })
                     } else {
-                        let start_bpos = self.last_pos - BytePos(2);
-                        while !self.curr_is('\n') && !self.is_eof() { self.bump(); }
-                        return Some(TokenAndSpan {
+                        Some(TokenAndSpan {
                             tok: token::Comment,
                             sp: codemap::mk_sp(start_bpos, self.last_pos)
```
Comment (alexcrichton) [inline src/libsyntax/parse/lexer/mod.rs]: I think this `start_bpos` is off by one perhaps? It was originally `last-BytesPos(2)` but now I think it's `last-BytePos(3)`

**Comment by alexcrichton** [comment]
Thanks! Would it be possible to add a test for this as well?

**Comment by Eljay** [comment]
Ok the bytepos should now be consistent with how it was previously, and I've fixed up the error message change that broke some tests. I've also added a test to check that the newlines are stripped but not sure if this is the best way to do it.

**Comment by alexcrichton** [comment]
Hm I believe though that in the first case (a doc comment) the byte pos was subtracted by 3, and in this case it's now subtracted by only two? Perhaps some more assertions could be added to the test that was added about the spans being returned as well?

**Comment by Eljay** [comment]
Previously the byte pos was `pos - 3` for doc comments and `last_pos - 2` otherwise to account for characters wider than 1 byte in comments (e.g. starting with `//<wide char>`). So `last_pos - 2` should be correct independent of the byte width of the current character. Correct?

**Comment by alexcrichton** [comment]
The branch for doc comments has `last_pos - 3` previously, and the branch for regular comments had `last_pos - 2`, but now both branches use `last_pos - 2` which is all I'm worried about

**Comment by Eljay** [comment]
It previously used `pos - 3`, not `last_pos - 3`. Seeing as the third character for doc comments is always `/` or `!`, which both have a byte width of 1, `last_pos - 2` should always be equal to `pos - 3` unless I'm going mad. I will make it explicit anyway to be sure.

**Comment by alexcrichton** [comment]
Oh wow I totally missed the `pos` vs `last_pos` distinction, this is subtle! Anyway I think current current version looks good to me, so thanks!

@bors: r+ 2dcc200

**Pull request merged**